actionpack/lib/action_dispatch/middleware/exception_wrapper.rb

```diff
@@ -148,15 +148,20 @@ def traces
       application_trace_with_ids = []
       framework_trace_with_ids = []
       full_trace_with_ids = []
+      application_traces = application_trace.map(&:to_s)
 
+      full_trace = backtrace_cleaner&.clean_locations(backtrace, :all) || backtrace
```

Comment:
If there are no locations, it will return an empty array, not nil:
https://edgeapi.rubyonrails.org/classes/ActiveSupport/BacktraceCleaner.html#method-i-clean_locations